web-common/src/features/custom-dashboards/AddChartMenu.svelte

```diff
@@ -1,23 +1,24 @@
 <script lang="ts">
+  import { useAllFileNames } from "@rilldata/web-common/features/entity-management/file-selectors";
```

Comment:
**Issue**: The addition of 'useAllFileNames' for fetching file names introduces a broader scope than the original 'useChartFileNames', potentially including files that are not charts. This could lead to unintended files being listed in the chart addition menu. <br> **Fix**: Implement a filtering mechanism to ensure only chart-related files are included in the 'chartFileNames' array. This could involve filtering based on file metadata or adopting a naming convention for chart files. <br> **Code Suggestion**: 
 ```
 -  $: allNamesQuery = useAllFileNames(queryClient, $runtime.instanceId);
 +  $: allNamesQuery = useAllFileNames(queryClient, $runtime.instanceId).then(files => files.filter(file => isChartFile(file)));
 ```

Reply:
I (real Eric) agree with this. This "AddChart" button should only surface valid charts. What about using `ListResources()` and filtering for charts?